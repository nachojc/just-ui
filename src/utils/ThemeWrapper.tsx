import { ReactNode } from "react";
import { ThemeProvider } from "styled-components";
import theme from "../styles/themes";

type ThemeName = keyof typeof theme;

type options = {
  themeName: ThemeName;
};
export const wrapWithTheme = (children: ReactNode, options?: options) => {
  const opt: options = { ...{ themeName: "defaultTheme" }, ...options };
  return <ThemeProvider theme={theme[opt.themeName]}>{children}</ThemeProvider>;
};
