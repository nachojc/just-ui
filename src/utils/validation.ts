export const hasText = (value?: string): boolean => {
  return typeof value === "string" && value?.trim().length > 0;
};
