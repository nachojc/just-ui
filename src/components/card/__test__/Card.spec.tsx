import { render } from "@testing-library/react";
import { Card } from "..";
import React from "react";
import { wrapWithTheme } from "../../../utils/ThemeWrapper";

describe("Card", () => {
  it("render", async () => {
    const { asFragment } = render(wrapWithTheme(<Card />));
    const btn = asFragment();
    expect(btn).toMatchSnapshot();
  });
  it("render with another Theme", async () => {
    const { asFragment } = render(wrapWithTheme(<Card />, { themeName: "publicTheme" }));
    const btn = asFragment();
    expect(btn).toMatchSnapshot();
  });
});
