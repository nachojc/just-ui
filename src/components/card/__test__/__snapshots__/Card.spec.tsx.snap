// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card render 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb irbxqa"
  >
    <div />
  </div>
</DocumentFragment>
`;

exports[`Card render with another Theme 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb irbxqa"
  >
    <div />
  </div>
</DocumentFragment>
`;
